### PR TITLE
Adds lightning fabric integration

### LINKED ISSUE #742: integrations: lightning fabric
Add a logger for lightning fabric (a lightweight wrapper around pytorch and theoretically other frameworks that's somewhere between raw pytorch and full pytorch-lightning): https://lightning.ai/docs/fabric/stable/api/loggers.html

### PR DESCRIPTION
**Pull request by dberenbaum**
Closes #742. Fabric is Pytorch Lightning's attempt to abstract out the lightweight parts of their codebase so users can benefit from those without having to entirely adapt Lightning.

- Example notebook is included: https://github.com/iterative/dvclive/blob/38d1f40843be5adc49ed718884f04496a7f376ff/examples/DVCLive-Fabric.ipynb
- Docs PR: https://github.com/iterative/dvc.org/pull/5039

The existing Lightning logger was updated to inherit from this one. To make everything work with the existing Lightning logger, I added a `sync` method that decouples `next_step` from saving all the info to Studio, dvc.yaml, etc. I made it public, but no strong opinion on whether it should be public or documented.

See the Tensorboard logger in lightning for an example of another logger:
- [Docs](https://lightning.ai/docs/fabric/stable/api/generated/lightning.fabric.loggers.TensorBoardLogger.html#lightning.fabric.loggers.TensorBoardLogger)
- [Fabric logger](https://github.com/Lightning-AI/pytorch-lightning/blob/master/src/lightning/fabric/loggers/tensorboard.py)
- [Lightning logger](https://github.com/Lightning-AI/pytorch-lightning/blob/master/src/lightning/pytorch/loggers/tensorboard.py)

### CODE REVIEW

File: examples/DVCLive-Fabric.ipynb [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "85qErT5yTEbN"
+      },
+      "outputs": [],
+      "source": [
+        "import argparse\n",
```
Comment (shcheklein) [inline examples/DVCLive-Fabric.ipynb:81]: seems to be unused?

File: examples/DVCLive-Fabric.ipynb [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+        "import torchvision.transforms as T\n",
+        "from lightning.fabric import Fabric, seed_everything\n",
+        "from lightning.fabric.utilities.rank_zero import rank_zero_only\n",
+        "from torch.optim.lr_scheduler import StepLR\n",
+        "from torchmetrics.classification import Accuracy\n",
+        "from torchvision.datasets import MNIST\n",
+        "\n",
+        "from dvclive.fabric import DVCLiveLogger\n",
+        "\n",
+        "DATASETS_PATH = (\"Datasets\")"
```
Comment (shcheklein) [inline examples/DVCLive-Fabric.ipynb:98]: review: `ipywidgets` was required for me by tqdm to run this cell

**Comment by shcheklein** [comment]
Great work! Thanks, @dberenbaum for pushing forward integrations. Some comments as I'm reviewing / running it:

- While running and after an experiment I can't see plots in VS Code (while it's visible in Jupyter just fine)

**Comment by shcheklein** [comment]
@dberenbaum looks good to me overall (I didn't review really deep the Fabric existing loggers to actually deeply review the implementation, but overall it looks fine). I've enabled mypy back + left a comment about VS Code to review, probably also Studio needs to be reviewed / tried out.

**Comment by dberenbaum** [comment]
Should mypy be included in https://github.com/iterative/dvclive/blob/main/.pre-commit-config.yaml?

**Comment by shcheklein** [comment]
I would ask @skshetry and the team on what the current state of the art for this :)

**Comment by dberenbaum** [comment]
@skshetry How should mypy checks be handled?

**Comment by dberenbaum** [comment]
ping @skshetry

**Comment by dberenbaum** [comment]
I'm able to see it in VS Code and Studio:

https://github.com/iterative/dvclive/assets/2308172/ce55849b-6825-4704-b934-ab85234f464f

https://github.com/iterative/dvclive/assets/2308172/dec278b2-f290-42de-b169-e49c22979389

**Comment by skshetry** [comment]
> I would ask @skshetry and the team on what the current state of the art for this :)

We use `nox` for environment management on py-template based projects. So, you can `pip install nox` and then `nox -s lint` will take care of everything related to linting.

Since dvclive seems to have removed nox, you can try adding mypy to pre-commit but up to you.
Look into dvc's pre-commit config.

**Comment by dberenbaum** [comment]
Depends on #757 now. The remaining test failure is unrelated.

**Pull request merged**